app/components/CreateSDBoxForm/validator.js

```diff
@@ -57,14 +57,11 @@ const validateUserGroupPermissions = (permission, index, errors) => {
 
 const validateIamRolePermissions = (permission, index, errors) => {
     errors.iamRolePermissions[`${index}`] = {}
-    if (! permission.accountId) {
-        errors.iamRolePermissions[`${index}`].accountId = 'Required'
-    } else if (! /^\d+$/.test(permission.accountId)) {
-        errors.iamRolePermissions[`${index}`].accountId = 'Numeric only'
-    }
 
-    if (! permission.iamRoleName) {
-        errors.iamRolePermissions[`${index}`].iamRoleName = 'You must enter a IAM role name for this permission'
+    if (! permission.iamPrincipalArn) {
+        errors.iamRolePermissions[`${index}`].iamPrincipalArn = 'You must enter a IAM principal ARN for this permission'
+    } else if (! /^arn:aws:(iam|sts)::.+$/.test(permission.iamPrincipalArn)) {
```

Comment:
Pull request looks good to me.   My only question is how did you get this regex?  Do we know this is right?  Can we test it somehow?

It seems like it is good to have some regex here but just curious if it is flexible enough to allow everything that should be allowed, etc.